### PR TITLE
Doc: Intro Update

### PR DESCRIPTION
**Pull request by ax3l**
Updated the intro paragraphs with ideas we discussed in the weekly meeting.

### CODE REVIEW

**Comment by jlvay** [comment]
Looks good! A couple of comments/suggestions:
- remove "is" in "ImpactX is runs"
- perhaps add "(with time as the independent variable)" after "general particle-in-cell code"?

**Comment by ax3l** [comment]
Thanks, updated :)

**Review by cemitch99 — COMMENTED** [review]
This looks good to me.  I suggest that there is a follow-up paragraph reproducing some of the text from the IMPACT-Z description (most importantly, examples of applications), an stating explicitly that tracking methods are symplectic.  I can contribute this if we agree on where it should be located.

**Comment by ax3l** [comment]
Thanks, @cemitch99! Oh yes, I forgot that we wanted to reuse text from the readme in https://github.com/impact-lbl/IMPACT-Z.

Feel free to contribute here or we merge and do a follow-up PR to add more. I think in the `As a beam dynamics code, ImpactX ...` could be a good paragraph to extend with applications and symplectic properties.

File: docs/source/index.rst [hunk 1]
```diff
@@ -3,23 +3,32 @@
 ImpactX
 -------
 
-ImpactX is an s-based beam dynamics code including space charge effects.
+ImpactX provides high-performance modeling of beam dynamics in particle accelerators with collective effects.
+
 This is the next generation of the `IMPACT-Z <https://github.com/impact-lbl/IMPACT-Z>`__ code.
+ImpactX runs on modern GPUs or CPUs alike, provides user-friendly interfaces suitable for AI/ML workflows, has many :ref:`benchmarks <usage-examples>` to ensure its correctness, and an extensive documentation.
+
+As a beam dynamics code, ImpactX uses the reference trajectory :math:`s` as the independent variable of motion to achieve large speedups.
```
Comment (cemitch99) [inline docs/source/index.rst]:
```suggestion
As a beam dynamics code, ImpactX uses the reference trajectory path length :math:`s` as the independent variable of motion to achieve large speedups.
The code includes the effects of externally applied fields from magnets and accelerating cavities as well as the effect of self-fields (space charge fields, CSR, wakefields, ...).
All particle tracking models are symplectic, and space charge is included by solving the Poisson equation in the beam rest frame.
The code may be used to model the dynamics of beams in both linear and ring accelerators.
```

File: docs/source/index.rst [hunk 2]
```diff
@@ -3,23 +3,35 @@
 ImpactX
 -------
 
-ImpactX is an s-based beam dynamics code including space charge effects.
+ImpactX provides high-performance modeling of beam dynamics in particle accelerators with collective effects.
```
Comment (ax3l) [inline docs/source/index.rst]:
```suggestion
ImpactX enables high-performance modeling of beam dynamics in particle accelerators with collective effects.
```

File: docs/source/index.rst [hunk 3]
```diff
@@ -3,23 +3,35 @@
...
+
+As a beam dynamics code, ImpactX uses the reference trajectory path length :math:`s` as the independent variable of motion to achieve large speedups.
+The code includes the effects of externally applied fields from magnets and accelerating cavities as well as the effect of self-fields (space charge fields, CSR, wakefields, ...).
+All particle tracking models are symplectic, and space charge is included by solving the Poisson equation in the beam rest frame.
+The code may be used to model the dynamics of beams in both linear and ring accelerators.
+See our :ref:`theory chapter <theory-concepts>` for details on our models, assumptions and concepts.
+
+ImpactX is part of the `Beam, Plasma & Accelerator Simulation Toolkit (BLAST) <https://blast.lbl.gov>`__.
+If you are looking for a general particle-in-cell code (with time as the independent variable) on GPUs and CPUs, take a look at `WarpX <https://warpx.readthedocs.io>`__.
+If you are looking for a specialized code for efficient *plasma wakefield* modeling on GPUs and CPUs, take a look at the quasi-static code `HiPACE++ <https://hipace.readthedocs.io>`__.
```
Comment (ax3l) [inline docs/source/index.rst]:
```suggestion
ImpactX is part of the `Beam, Plasma & Accelerator Simulation Toolkit (BLAST) <https://blast.lbl.gov>`__.
Please see this page for more information to select the code most appropriate for your application.
```

**Pull request merged**